builtin/providers/azurerm/resource_arm_redis_cache.go

```diff
@@ -0,0 +1,445 @@
+
+	capacity := int32(d.Get("capacity").(int))
+	family := redis.SkuFamily(d.Get("family").(string))
+	sku := redis.SkuName(d.Get("sku_name").(string))
+
+	tags := d.Get("tags").(map[string]interface{})
+	expandedTags := expandTags(tags)
+
+	parameters := redis.UpdateParameters{
+		UpdateProperties: &redis.UpdateProperties{
+			EnableNonSslPort: &enableNonSSLPort,
+			Sku: &redis.Sku{
+				Capacity: &capacity,
+				Family:   family,
+				Name:     sku,
+			},
+			Tags: expandedTags,
+		},
+	}
+
+	if v, ok := d.GetOk("shard_count"); ok {
+		if d.HasChange("shard_count") {
+			shardCount := int32(v.(int))
+			parameters.ShardCount = &shardCount
+		}
+	}
+
+	if d.HasChange("redis_configuration") {
+		redisConfiguration := expandRedisConfiguration(d)
+		parameters.RedisConfiguration = redisConfiguration
+	}
+
+	_, err := client.Update(resGroup, name, parameters, make(chan struct{}))
+	if err != nil {
+		return err
+	}
+
+	read, err := client.Get(resGroup, name)
+	if err != nil {
+		return err
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read Redis Instance %s (resource group %s) ID", name, resGroup)
+	}
+
+	log.Printf("[DEBUG] Waiting for Redis Instance (%s) to become available", d.Get("name"))
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"Updating", "Creating"},
+		Target:     []string{"Succeeded"},
+		Refresh:    redisStateRefreshFunc(client, resGroup, name),
+		Timeout:    60 * time.Minute,
+		MinTimeout: 15 * time.Second,
+	}
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf("Error waiting for Redis Instance (%s) to become available: %s", d.Get("name"), err)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmRedisCacheRead(d, meta)
+}
+
+func resourceArmRedisCacheRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).redisClient
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["Redis"]
+
+	resp, err := client.Get(resGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error making Read request on Azure Redis Cache %s: %s", name, err)
+	}
+	if resp.StatusCode == http.StatusNotFound {
+		d.SetId("")
+		return nil
+	}
+
+	keysResp, err := client.ListKeys(resGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error making ListKeys request on Azure Redis Cache %s: %s", name, err)
+	}
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resGroup)
+	d.Set("location", azureRMNormalizeLocation(*resp.Location))
+	d.Set("ssl_port", resp.SslPort)
+	d.Set("host_name", resp.HostName)
+	d.Set("port", resp.Port)
+	d.Set("enable_non_ssl_port", resp.EnableNonSslPort)
+	d.Set("capacity", resp.Sku.Capacity)
+	d.Set("family", resp.Sku.Family)
+	d.Set("sku_name", resp.Sku.Name)
+
+	if resp.ShardCount != nil {
+		d.Set("shard_count", resp.ShardCount)
+	}
+
+	redisConfiguration := flattenRedisConfiguration(resp.RedisConfiguration)
+	d.Set("redis_configuration", &redisConfiguration)
```

Comment:
this can be refactored a little to be something like:

```
if err := d.Set("redis_configuration", flattenRedisConfiguration(resp.RedisConfiguration)); err != nil {
    return fmt.Errorf("[DEBUG] Error setting redis_configuration error: %#v", err)
}
```

Please pay attention to the fact that I have changed the d.Set from receiving a pointer